<a href="https://colab.research.google.com/github/MarcosArAr/Sales-predictions/blob/main/Proyecto_1_Parte_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Data cleaning

In [72]:
import pandas as pd
df = pd.read_csv("/content/sales_predictions (1).csv")
df

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0              FDA15        9.300          Low Fat         0.016047   
1              DRC01        5.920          Regular         0.019278   
2              FDN15       17.500          Low Fat         0.016760   
3              FDX07       19.200          Regular         0.000000   
4              NCD19        8.930          Low Fat         0.000000   
...              ...          ...              ...              ...   
8518           FDF22        6.865          Low Fat         0.056783   
8519           FDS36        8.380          Regular         0.046982   
8520           NCJ29       10.600          Low Fat         0.035186   
8521           FDN46        7.210          Regular         0.145221   
8522           DRG01       14.800          Low Fat         0.044878   

                  Item_Type  Item_MRP Outlet_Identifier  \
0                     Dairy  249.8092            OUT049   
1               Soft Drinks   48.2692            OUT018   
2                      Meat  141.6180            OUT049   
3     Fruits and Vegetables  182.0950            OUT010   
4                 Household   53.8614            OUT013   
...                     ...       ...               ...   
8518            Snack Foods  214.5218            OUT013   
8519           Baking Goods  108.1570            OUT045   
8520     Health and Hygiene   85.1224            OUT035   
8521            Snack Foods  103.1332            OUT018   
8522            Soft Drinks   75.4670            OUT046   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                          1999      Medium               Tier 1   
1                          2009      Medium               Tier 3   
2                          1999      Medium               Tier 1   
3                          1998         NaN               Tier 3   
4                          1987        High               Tier 3   
...                         ...         ...                  ...   
8518                       1987        High               Tier 3   
8519                       2002         NaN               Tier 2   
8520                       2004       Small               Tier 2   
8521                       2009      Medium               Tier 3   
8522                       1997       Small               Tier 1   

            Outlet_Type  Item_Outlet_Sales  
0     Supermarket Type1          3735.1380  
1     Supermarket Type2           443.4228  
2     Supermarket Type1          2097.2700  
3         Grocery Store           732.3800  
4     Supermarket Type1           994.7052  
...                 ...                ...  
8518  Supermarket Type1          2778.3834  
8519  Supermarket Type1           549.2850  
8520  Supermarket Type1          1193.1136  
8521  Supermarket Type2          1845.5976  
8522  Supermarket Type1           765.6700  

[8523 rows x 12 columns]

In [42]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


In [43]:
df.describe()

Item_Weight  Item_Visibility     Item_MRP  Outlet_Establishment_Year  \
count  7060.000000      8523.000000  8523.000000                8523.000000   
mean     12.857645         0.066132   140.992782                1997.831867   
std       4.643456         0.051598    62.275067                   8.371760   
min       4.555000         0.000000    31.290000                1985.000000   
25%       8.773750         0.026989    93.826500                1987.000000   
50%      12.600000         0.053931   143.012800                1999.000000   
75%      16.850000         0.094585   185.643700                2004.000000   
max      21.350000         0.328391   266.888400                2009.000000   

       Item_Outlet_Sales  
count        8523.000000  
mean         2181.288914  
std          1706.499616  
min            33.290000  
25%           834.247400  
50%          1794.331000  
75%          3101.296400  
max         13086.964800

In [44]:
df['Outlet_Size'].value_counts()

Medium    2793
Small     2388
High       932
Name: Outlet_Size, dtype: int64

In [45]:
df["Outlet_Type"].value_counts()

Supermarket Type1    5577
Grocery Store        1083
Supermarket Type3     935
Supermarket Type2     928
Name: Outlet_Type, dtype: int64

## Columna Item_Weight

In [46]:
df['Item_Weight'].value_counts()

12.150    86
17.600    82
13.650    77
11.800    76
15.100    68
          ..
7.275      2
7.685      1
9.420      1
6.520      1
5.400      1
Name: Item_Weight, Length: 415, dtype: int64

In [47]:
df['Item_Weight'].mode()

0    12.15
Name: Item_Weight, dtype: float64

In [48]:
df['Item_Weight'].mean()

12.857645184135976

In [49]:
df['Item_Weight'].median()

12.6

In [79]:
df['Item_Weight'].fillna(12.8, inplace = True)

In [51]:
df['Item_Weight'].median()

12.8

In [52]:
df['Item_Weight'].mean()

12.847750205326763

In [53]:
df['Item_Weight'].mode()

0    12.8
Name: Item_Weight, dtype: float64

In [54]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                8523 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


## Columna "Outlet_Size"

#### De los 2410 valores nulos en la columna 'Outlet_Size', se evaluara por cada tipo.

##### Todos los 'Supermarket Type2' son de tipo 'Medium'



In [55]:
df_type2 = df["Outlet_Type"] == "Supermarket Type2"
df_size2 = df["Outlet_Size"] == "Medium"
df[df_type2 & df_size2]

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
1              DRC01        5.920          Regular         0.019278   
5              FDP36       10.395          Regular         0.000000   
16             NCB42       11.800          Low Fat         0.008596   
31             NCS17       18.600          Low Fat         0.080829   
32             FDP33       18.700          Low Fat         0.000000   
...              ...          ...              ...              ...   
8506           DRF37       17.250          Low Fat         0.084676   
8511           FDF05       17.500          Low Fat         0.026980   
8515           FDH24       20.700          Low Fat         0.021518   
8516           NCJ19       18.600          Low Fat         0.118661   
8521           FDN46        7.210          Regular         0.145221   

               Item_Type  Item_MRP Outlet_Identifier  \
1            Soft Drinks   48.2692            OUT018   
5           Baking Goods   51.4008            OUT018   
16    Health and Hygiene  115.3492            OUT018   
31    Health and Hygiene   96.4436            OUT018   
32           Snack Foods  256.6672            OUT018   
...                  ...       ...               ...   
8506         Soft Drinks  263.1910            OUT018   
8511        Frozen Foods  262.5910            OUT018   
8515        Baking Goods  157.5288            OUT018   
8516              Others   58.7588            OUT018   
8521         Snack Foods  103.1332            OUT018   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
1                          2009      Medium               Tier 3   
5                          2009      Medium               Tier 3   
16                         2009      Medium               Tier 3   
31                         2009      Medium               Tier 3   
32                         2009      Medium               Tier 3   
...                         ...         ...                  ...   
8506                       2009      Medium               Tier 3   
8511                       2009      Medium               Tier 3   
8515                       2009      Medium               Tier 3   
8516                       2009      Medium               Tier 3   
8521                       2009      Medium               Tier 3   

            Outlet_Type  Item_Outlet_Sales  
1     Supermarket Type2           443.4228  
5     Supermarket Type2           556.6088  
16    Supermarket Type2          1621.8888  
31    Supermarket Type2          2741.7644  
32    Supermarket Type2          3068.0064  
...                 ...                ...  
8506  Supermarket Type2          3944.8650  
8511  Supermarket Type2          4207.8560  
8515  Supermarket Type2          1571.2880  
8516  Supermarket Type2           858.8820  
8521  Supermarket Type2          1845.5976  

[928 rows x 12 columns]

##### Todos los 'Supermarket Type3' son de tipo 'Medium'

In [56]:
df_type3 = df["Outlet_Type"] == "Supermarket Type3"
df_size3 = df["Outlet_Size"] == "Medium"
df[df_type3 & df_size3]

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
7              FDP10         12.8          Low Fat         0.127470   
18             DRI11         12.8          Low Fat         0.034238   
21             FDW12         12.8          Regular         0.035400   
36             FDV20         12.8          Regular         0.059512   
38             FDX10         12.8          Regular         0.123111   
...              ...          ...              ...              ...   
8472           NCS41         12.8          Low Fat         0.053185   
8485           DRK37         12.8          Low Fat         0.043792   
8487           DRG13         12.8          Low Fat         0.037006   
8488           NCN14         12.8          Low Fat         0.091473   
8504           NCN18         12.8          Low Fat         0.124111   

                  Item_Type  Item_MRP Outlet_Identifier  \
7               Snack Foods  107.7622            OUT027   
18              Hard Drinks  113.2834            OUT027   
21             Baking Goods  144.5444            OUT027   
36    Fruits and Vegetables  128.0678            OUT027   
38              Snack Foods   36.9874            OUT027   
...                     ...       ...               ...   
8472     Health and Hygiene  182.9608            OUT027   
8485            Soft Drinks  189.0530            OUT027   
8487            Soft Drinks  164.7526            OUT027   
8488                 Others  184.6608            OUT027   
8504              Household  111.7544            OUT027   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
7                          1985      Medium               Tier 3   
18                         1985      Medium               Tier 3   
21                         1985      Medium               Tier 3   
36                         1985      Medium               Tier 3   
38                         1985      Medium               Tier 3   
...                         ...         ...                  ...   
8472                       1985      Medium               Tier 3   
8485                       1985      Medium               Tier 3   
8487                       1985      Medium               Tier 3   
8488                       1985      Medium               Tier 3   
8504                       1985      Medium               Tier 3   

            Outlet_Type  Item_Outlet_Sales  
7     Supermarket Type3          4022.7636  
18    Supermarket Type3          2303.6680  
21    Supermarket Type3          4064.0432  
36    Supermarket Type3          2797.6916  
38    Supermarket Type3           388.1614  
...                 ...                ...  
8472  Supermarket Type3          1653.8472  
8485  Supermarket Type3          6261.8490  
8487  Supermarket Type3          4111.3150  
8488  Supermarket Type3          2756.4120  
8504  Supermarket Type3          4138.6128  

[935 rows x 12 columns]

##### Todos los 'Grocery' son de tipo 'Small'. Los 'Grocery Store' presentan 555 valores nulos en la columna tipo, estos seran reemplazados por el tipo 'Small'.

In [58]:
df_typegrocery = df[df["Outlet_Type"] == "Grocery Store"]
df_typegrocery["Outlet_Size"].value_counts()

Small    528
Name: Outlet_Size, dtype: int64

In [59]:
df_typegrocery = df[df["Outlet_Type"] == "Grocery Store"]
df_typegrocery["Outlet_Size"].isna().value_counts()

True     555
False    528
Name: Outlet_Size, dtype: int64

In [73]:
df_filter54 = df["Outlet_Type"] == "Grocery Store"
df_filter55 = df["Outlet_Size"].isna()
df.loc[df_filter54 & df_filter55,"Outlet_Size"] = "Small"

In [75]:
df["Outlet_Size"].isna().value_counts()

False    6668
True     1855
Name: Outlet_Size, dtype: int64

##### La columan 'Supermarket Type1' contienen 1855 valores nulos, asi mismo presenta diferentes tipos de tamaños ('Medium', 'Small', 'High').

In [62]:
df_type1 = df[df["Outlet_Type"] == "Supermarket Type1"]
df_type1["Outlet_Size"].value_counts()

Small     1860
High       932
Medium     930
Name: Outlet_Size, dtype: int64

In [63]:
df_type1["Outlet_Size"].isna().value_counts()

False    3722
True     1855
Name: Outlet_Size, dtype: int64

In [64]:
df["Outlet_Size"].mode()

0    Small
Name: Outlet_Size, dtype: object

In [65]:
df["Outlet_Size"].value_counts()

Small     2943
Medium    2793
High       932
Name: Outlet_Size, dtype: int64

In [66]:
df_type1["Outlet_Size"].mode()

0    Small
Name: Outlet_Size, dtype: object

In [76]:
# Los valores faltantes de la columna Outlet_Size pertenencientes a SupermarketType1 seran reemplazados por la moda la columna Outlet_Size "Small"
df_filter57 = df["Outlet_Type"] == "Supermarket Type1"
df_filter58 = df["Outlet_Size"].isna()
df.loc[df_filter57 & df_filter58,"Outlet_Size"] = "Small"

In [77]:
df["Outlet_Size"].isna().value_counts()

False    8523
Name: Outlet_Size, dtype: int64

## Data limpio

In [80]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                8523 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                8523 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB
